### DISCUSSION TITLE
Auth for Tesla

### DISCUSSION THREAD

**Discussion by liveleeiow** [alandtse/tesla#793] [Q&A]
Hi. I have this app and the access token token is saying it's only valid for 8 hours....Is this how long I have to use the token for the 1st time?

Once i've used it how frequently does the token need to be updated in HA?

Thank you.

**Comment by alandtse** [comment]
We ask for the refresh token which will automatically generate the required tokens. However, it's Tesla's call on how long anything works so it could be forever or a minute after you login. This is an unofficial integration and Tesla is actively planning to break the unofficial API we are using.

**Reply by DrakahnFb** [reply]
Hi Alan, it seems that Tesla stopped the unoffial API and opened a new API, the [FleetAPI](https://developer.tesla.com/docs/fleet-api). There they write:
FleetAPI is a RESTful data and command service providing access to Tesla vehicles and energy devices. Partners can interact with their own devices, or devices for which they have been granted access by a customer.
Do you plan to adept your Tesla Custom Integration to the new API?

**Reply by skipishere** [reply]
That's a lot of discussion around the API in #774 and #743

**Comment by liveleeiow** [comment]
Ok, thank you for the update.  Hopefully we have a long time of it working.

Thank you for the work in making this for the community.
